packages/react-noop-renderer/src/ReactNoopShared.js

```diff
@@ -0,0 +1,582 @@
+/**
```

Comment:
For some reason I'm not a big fan of "Shared" as a name. It's like utils. Doesn't say anything. Don't really know where this is going yet so hard to find a better name.